### PR TITLE
net-misc/olsrd: Fix incompatible pointer to integer conversion initia…

### PR DESCRIPTION
**Pull request by listout**
…lizing int

Closes: https://bugs.gentoo.org/898090

### CODE REVIEW

**Comment by ceamac** [comment]
Doesn't compile with clang 16 on musl.

[build.log.txt](https://github.com/gentoo/gentoo/files/11703595/build.log.txt)
[emerge-info.txt](https://github.com/gentoo/gentoo/files/11703600/emerge-info.txt)

It's the same log as the one in the bug, actually.  I think your patch was meant to fix this bug instead https://bugs.gentoo.org/870748

File: net-misc/olsrd/files/olsrd-0.9.8-clang16-buildfix.patch [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+ 
+ #include <stdio.h>
+ #include <string.h>
+@@ -80,6 +81,10 @@ struct filter_group {
+ 
+ static struct filter_group * filter_groups = NULL;
+ 
++// Delcaring function prototype here, fixes build error with clang-16
++// Refer: https://bugs.gentoo.org/898090
++int should_filter(union olsr_ip_addr * originator);
```
Comment (thesamesam) [inline net-misc/olsrd/files/olsrd-0.9.8-clang16-buildfix.patch:23]: But why? This needs more justification.

Keep in mind that for _all_ of these bugs, there's always a wrong fix and a right fix:
- a cast (sometimes it's appropriate, but keep in mind, it's just effectively silencing the warning)
- passing the right argument (which may involve a cast, but not always, but a cast should not be done blindly)

Comment (thesamesam) [reply net-misc/olsrd/files/olsrd-0.9.8-clang16-buildfix.patch:23]: Ah, I see _this_ part (the redeclaration) is for an implicit. func. decl error, which is OK. But my above point still stands wrt casts and fixing these bugs in general.

Comment (listout) [reply net-misc/olsrd/files/olsrd-0.9.8-clang16-buildfix.patch:23]: I don't think this was related to -Wincompatible-function-pointer-types, but rather -Werror=implicit-function-declaration

Comment (listout) [reply net-misc/olsrd/files/olsrd-0.9.8-clang16-buildfix.patch:23]: > Doesn't compile with clang 16 on musl.

Thanks for the report, I'll fix it (don't have access to musl system right now).

**Comment by listout** [comment]
> Doesn't compile with clang 16 on musl.

Thanks for the report, I'll fix it (don't have access to musl system right now).

**Review by ceamac — APPROVED** [review]
I tested it on both musl and glibc, seems fine.  Thank you!

I'll add https://bugs.gentoo.org/870748 to closed bugs.

**Pull request closed without merging**